### PR TITLE
Change email signup to disabled after clicking

### PR DESCRIPTION
**Pull request by johnduffell**
## Why are you doing this?
Sometimes when network issues happened, we were getting multiple duplicate requests for the sign up to the emails button.

This was because when they clicked the button it wasn't actually disabled and the visual appearance didn't change, so during slow network conditions they would click it quite a few times.

This PR changes the button to be disabled when they click, until the resonse comes in.
It also changes the button to look different so people know their click was registered.

[**Trello Card**](https://trello.com/c/IC361x2a/198-investigate-500-errors-should-they-be-500s)

before clicking (no change)
image
new - while loading (used to be the same as before clicking)
image
after complete (no change)
image
after error (no change)
image

@jranks123 for tech review and @ionamckendrick for behaviour review please

### CODE REVIEW

**Comment by johnduffell** [comment]
not sure why the build is failing, as it is not true, so some linter help would be great!
```
/Users/jduffell/ws/support-frontend/assets/components/marketingConsentNew/marketingConsent.jsx
  22:9   error  'csrf' PropType is defined but prop is never used     react/no-unused-prop-types
  23:12  error  'onClick' PropType is defined but prop is never used  react/no-unused-prop-types
  24:10  error  'error' PropType is defined but prop is never used    react/no-unused-prop-types
  25:12  error  'loading' PropType is defined but prop is never used  react/no-unused-prop-types
```

**Comment by johnduffell** [comment]
thanks to @jranks123 this should build, and I've tested it again as working! 🌟 ready for a thumbs up.

**Comment by jranks123** [comment]
I'm not sure about `Loading`. Not sure that language fits. Rather, how about a spinner in the button?

File: assets/pages/regular-contributions/__tests__/__snapshots__/regularContributionsReducersTest.js.snap [hunk 1]
```diff
@@ -30,6 +30,7 @@ Object {
   "marketingConsent": Object {
     "confirmOptIn": null,
     "error": false,
+    "loading": false,
```
Comment (jranks123) [inline assets/pages/regular-contributions/__tests__/__snapshots__/regularContributionsReducersTest.js.snap]: I feel like loading is quite ambiguous here, maybe a better term would be `requestPending`

Comment (johnduffell) [reply assets/pages/regular-contributions/__tests__/__snapshots__/regularContributionsReducersTest.js.snap]: will do

**Comment by johnduffell** [comment]
I've updated everything to request pending now, plus the button
image

**Pull request merged**